### PR TITLE
 #161404710 Add authentication endpoints

### PR DESCRIPTION
**Pull request by kwanj-k**
### What does this PR do?
Add all the authentication related endpoints

#### Description of Task to be Completed?
-- Create a registration/signup endpoint
-- Create a login endpoint
-- Create endpoints for adding an admin/attendant

### How should this be manually tested?
* After cloning the repo, git pull the branch ft-auth-#161404710 
* Install a virtual enviroment and activate it [Refer here](https://docs.python-guide.org/dev/virtualenvs/)
* Pip install the the requirements.txt
* Run the following:
  ``` export FLASK_APP='run.py' ```
  ``` export APP_SETTINGS='development' ```
   ``` pytest --cov-report term-missing --cov=app ```

Also, you need to create a development and test database and connect to it.[Refer here](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-18-04#creating-a-new-role)

Then export the database URL in the environment:
```
export Dev_URL="dbname=dbname user=username password=password"
export Test_URL="dbname=dbname user=username password=password"
export SECRET="iamasecrettheunsadtheunsulliedtheundead"

```
We need the secret key for generating the tokens

Then Run the tests:
```
pytest --cov-report term-missing --cov=app
```
Four tests should now pass as shown:
screenshot from 2018-10-24 20-21-06

### What are the relevant Pivotal Tracker stories?
  #161404710

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,16 +2,27 @@ language: python
...
   - pytest --cov-report term-missing --cov=app
   - coveralls
 
 after_success:
-  - coveralls
\ No newline at end of file
+  - coveralls
+
+cache:
+  - pip
```
Comment (dotNesh) [inline .travis.yml]: At the end of the file please leave one empty line.

Comment (kwanj-k) [reply .travis.yml]: I will

File: .travis.yml [hunk 2]
```diff
@@ -14,4 +22,5 @@ script:
   - coveralls
 
 after_success:
-  - coveralls
\ No newline at end of file
+  - coveralls
+
```
Comment (dotNesh) [inline .travis.yml]: Is there a trailing whitespace here?

File: app/api/v2/db_config.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+This file contains the database configurations and common operations
+"""
+
+#Standard library imports
+import os
+import psycopg2
+
+
+config_name = os.getenv('APP_SETTINGS')
+durl = os.getenv('Dev_URL')
```
Comment (dotNesh) [inline app/api/v2/db_config.py]: Have a better naming for the url variables, one that can easily be understood.

File: app/api/v2/db_config.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    for q in queries:
+        cur.execute(q)
+    conn.commit()
+
+def drop_all():
+    """
+    Method to drop all the tables in the database
+    """
+    cur.execute("DROP TABLE IF EXISTS users CASCADE")
+    cur.execute("DROP TABLE IF EXISTS stores CASCADE")
```
Comment (dotNesh) [inline app/api/v2/db_config.py:38]: Does this mean you have only two tables?

File: app/api/v2/models/accounts.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                rank = 'SuperAdmin'
+            if self.role==1:
+                rank = 'Admin'
+            if self.role==2:
+                rank = 'Attendant'
+            return rank
+        return dict(
+            email=self.email,
+            role=rank(),
+            added_at = self.added_at)
```
Comment (dotNesh) [inline app/api/v2/models/accounts.py]: leave a blank space at the end of the file.

File: app/api/v2/routes.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+v_2 = Blueprint('v_2', __name__, url_prefix="/api/v2")
+api = Api(v_2)
+v2 = api.namespace(
+    'v2',
+    description='Store manager Api Using postgres',
+    authorizations=authorizations)
+
+api.add_namespace(newstore_route, path="/signup")
+api.add_namespace(login_route, path="/")
```
Comment (dotNesh) [inline app/api/v2/routes.py]: leave a blank space at the end of the file.

File: app/tests/v2/base_config.py [hunk 7]
```diff
@@ -9,20 +9,43 @@
 
 # Local application imports
 from app.apps import create_app
+from app.api.v2.db_config import create_tables,drop_all
 
 
 config_name = "testing"
 app = create_app(config_name)
 
+l_url = "/api/v2/auth/login"
```
Comment (dotNesh) [inline app/tests/v2/base_config.py]: Use clear names for your variables.

File: app/api/v2/db_config.py (hunk 3 above)
Comment (kwanj-k) [reply app/api/v2/db_config.py]: On it

File: app/api/v2/db_config.py (hunk 4 above)
Comment (kwanj-k) [reply app/api/v2/db_config.py:38]: Yes,for now..but I will also have products table,records and carts table

Comment (kwanj-k) [reply app/api/v2/db_config.py:38]: The next PR will contain all those tables, am working on the structure

Comment (dotNesh) [reply app/api/v2/db_config.py:38]: That's okay.

**Review by dotNesh — APPROVED** [review]
Great work. Looks good to me.

**Pull request merged**